adapters/silverpush/silverpush.go

```diff
@@ -0,0 +1,346 @@
+package silverpush
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	bidderConfig  = "sp_pb_ortb"
+	bidderVersion = "1.0.0"
+)
+
+type adapter struct {
+	bidderName string
+	endpoint   string
+}
+
+type SilverPushImpExt map[string]json.RawMessage
+
+type SilverPushReqExt struct {
+	PublisherId string  `json:"publisherId"`
+	BidFloor    float64 `json:"bidfloor"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "No impressions in bid request."}}
+	}
```

Comment:
You do not need to worry about this case. Prebid Server will not call your adapter if there are no impressions. Please remove this check and pull up `ValidateAndProcessRequest(request)` into this method.

Reply:
Fixed
[commit](https://github.com/prebid/prebid-server/pull/2957/commits/cc30407f798a60b4550738f3802ffe7a19f81493)